src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT INTO points (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
+            statement.setInt(1, point.getId());
+            statement.setDouble(2, point.getXCoordinate());
+            statement.setDouble(3, point.getYCoordinate());
+            statement.executeUpdate();
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                while (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    point.setId(id);
+                    System.out.println("Point inserted successfully. Generated ID: " + id);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public void deletePoint(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("DELETE FROM points WHERE id = ?")) {
+            statement.setInt(1, id);
+            statement.executeUpdate();
+            System.out.println("Point with ID " + id + " deleted successfully.");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+
+    @Override
+    public Point getPoint(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        Point point = null;
+        try (PreparedStatement statement = connection.prepareStatement("SELECT  xCoordinate, yCoordinate FROM points WHERE id = ?")) {
+            statement.setInt(1, id);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    double xCoordinate = resultSet.getDouble("xCoordinate");
+                    double yCoordinate = resultSet.getDouble("yCoordinate");
+                    point = new Point(id, xCoordinate, yCoordinate);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+        return point;
+    }
+
+    @Override
+    public List<Point> getPoints() {
+        List<Point> points = new ArrayList<>();
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("SELECT id, xCoordinate, yCoordinate FROM points");
```

Comment:
This one should be
SELECT p.id AS pointId, p.x_coordinate AS xCoordinate, p.y_coordinate AS yCoordinate FROM points p